### PR TITLE
New tracks api

### PR DESCRIPTION
**Pull request by j8seangel**
Includes new tracks API renderer.
- [x] Uses geojson as data source from the new [API](http://globalfishingwatch.io/api-docs/console/?url=https://vessels-dot-world-fishing-827.appspot.com/api#/default/get_datasets__dataset__vessels__vesselId__tracks). Needs a new flag on the track array with `type: 'geojson'`
- [x] Stop using pixi.js for tracks as it the geojson si filtered by hand and pass as a mapbox-gl style layer.
- [x] Fit to track bounds. This is done by including a new flag called `fitBoundsOnLoad` set to true
Note: removing [mapbox-extent](https://github.com/mapbox/geojson-extent) in flavour of [turf bbox](http://turfjs.org/docs/#bbox) because of this:
image

Bonus:
- Allows to [debug](https://github.com/GlobalFishingWatch/map-client/compare/as-module...new-tracks-api?expand=1#diff-00268b9b16f5b8b6dbe5689d9cf3294bR268) module store remotely.

image

### CODE REVIEW

File: app/src/_map/tracks/tracks.selectors.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+export const getTracksStyles = createSelector(
+  [getTemporalExtent, getTracksData],
+  (temporalExtent, tracks) => {
+    const hasTemporalExtent = temporalExtent && temporalExtent.length > 0;
+    const hasTracks = tracks && tracks.length > 0;
+    if (!hasTemporalExtent || !hasTracks) return null;
+
+    const timerange = {
+      start: temporalExtent[0].getTime(),
```
Comment (nerik) [inline app/src/_map/tracks/tracks.selectors.js]: I think we need to talk about UTC again 😰

File: app/src/_map/tracks/tracks.selectors.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            type: 'geojson',
+            data: filterGeojsonByTimerange(track.data, timerange)
+          }
+        },
+        layers: [
+          {
+            id: `${track.id}Lines`,
+            source,
+            type: 'line',
+            interactive: true,
```
Comment (nerik) [inline app/src/_map/tracks/tracks.selectors.js]: This parameter is now deprecated (and ignored) by Mapbox GL

File: app/src/_map/tracks/tracks.selectors.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            paint: {
+              'line-width': 1,
+              'line-color': track.color
+            }
+          },
+          {
+            id: `${track.id}Points`,
+            source,
+            type: 'circle',
+            interactive: true,
```
Comment (nerik) [inline app/src/_map/tracks/tracks.selectors.js]: This parameter is now deprecated (and ignored) by Mapbox GL

File: app/src/_map/tracks/tracks.selectors.js (hunk 2 above)
Comment (nerik) [reply app/src/_map/tracks/tracks.selectors.js]: Interaction on layers is now handled here https://github.com/GlobalFishingWatch/map-client/blob/as-module/app/src/_map/glmap/Map.container.js#L43

File: app/src/_map/tracks/tracks.selectors.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
+import { createSelector } from 'reselect';
+import { getTemporalExtent } from '../module/module.selectors.js';
+
+export const getTracksData = state => state.map.tracks.data;
+
+const filterGeojsonByTimerange = (geojson, { start, end }) => {
+  if (!geojson || !geojson.features) return null;
+  const featuresFiltered = geojson.features.reduce((acc, feature) => {
```
Comment (nerik) [inline app/src/_map/tracks/tracks.selectors.js]: Is it possible to rename `acc` to something more meaningful?

File: app/src/_map/tracks/tracks.selectors.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    const hasTimes = feature.properties.coordinateProperties.times && feature.properties.coordinateProperties.times.length > 0;
+    if (hasTimes) {
+      const filtered = feature.geometry.coordinates.reduce((acc2, coordinate, index) => {
```
Comment (nerik) [inline app/src/_map/tracks/tracks.selectors.js]: Is it possible to rename `acc2` to something more meaningful?

File: app/src/_map/README.md [hunk 6]
```diff
@@ -265,4 +265,11 @@ Function. TODO
...
 
 
+# Development
 
+As the module is using redux internally it is hard to debug the store or know which actions are being dispatched. To solve this you could run the [remode dedux devtools](https://github.com/zalmoxisus/remote-redux-devtools) with the following steps:
+
+1. Run the websockets server with `npm run debug-server`
+2. Follow https://github.com/zalmoxisus/remote-redux-devtools#monitoring to see the logs
+
+TODO: Include a `REDUX_REMOTE_DEBUG` env variable when moved to map-components
```
[the remark this answers is not in the document]
Comment (j8seangel) [reply app/src/_map/README.md]: I think it worth it to be included once we move the module to map-components as this ENV variable is not the scope of the project itself. So I'll create an issue to not forget it.

File: app/src/_map/tracks/tracks.selectors.js (hunk 2 above)
Comment (j8seangel) [reply app/src/_map/tracks/tracks.selectors.js]: Removed, thanks for the info!

File: app/src/_map/README.md [hunk 7]
```diff
@@ -265,4 +265,11 @@ Function. TODO
 Function. Notify of attributions changes depending on layers toggled [PARTIALLY IMPLEMENTED] Will only fire at start.
 
 
+# Development
 
+As the module is using redux internally it is hard to debug the store or know which actions are being dispatched. To solve this you could run the [remode dedux devtools](https://github.com/zalmoxisus/remote-redux-devtools) with the following steps:
```
[the remark this answers is not in the document]
Comment (j8seangel) [reply app/src/_map/README.md]: 🙈

File: app/src/_map/README.md [hunk 8]
```diff
@@ -234,11 +234,11 @@ React Node. Mandatory. DOM Node to display inside popup.
 
 ### `clickPopup.latitude`
 
-Number. Mandatory. Latitude for anchor point. 
+Number. Mandatory. Latitude for anchor point.
 
 ### `clickPopup.longitude`
 
-Number. Mandatory. Longitude for anchor point. 
+Number. Mandatory. Longitude for anchor point.
```
[the remark this answers is not in the document]
Comment (j8seangel) [reply app/src/_map/README.md]: Added to the documentation 📝

File: app/src/_map/tracks/tracks.reducer.js [hunk 9]
```diff
@@ -1,41 +1,63 @@
...
+    type: PropTypes.oneOf(['geojson', undefined]),
+    color: PropTypes.string,
+    data: PropTypes.object,
+    geoBounds: PropTypes.exact({
+      minLat: PropTypes.number,
+      minLng: PropTypes.number,
+      maxLat: PropTypes.number,
+      maxLng: PropTypes.number
+    }),
+    fitBoundsOnLoad: PropTypes.bool,
```
[the remark this answers is not in the document]
Comment (j8seangel) [reply app/src/_map/tracks/tracks.reducer.js]: Indeed, so I've created a new `proptypes` folder with the module proptypes definition.
In that way, we can use and extend when needed in reducers without duplications.
See example: https://github.com/GlobalFishingWatch/map-client/commit/9f6bd765d923206aa9a2925acab43848dd9bd4ad

File: app/src/_map/tracks/tracks.selectors.js (hunk 4 above)
Comment (j8seangel) [reply app/src/_map/tracks/tracks.selectors.js]: Good catch, fixed here: https://github.com/GlobalFishingWatch/map-client/commit/15eb1b14930dc18c83d63e09fa17cfff1887a33a

File: app/src/_map/tracks/tracks.selectors.js (hunk 1 above)
Comment (nerik) [reply app/src/_map/tracks/tracks.selectors.js]: https://github.com/GlobalFishingWatch/GFW-Tasks/issues/986

File: app/src/_map/index.jsx [hunk 10]
```diff
@@ -1,15 +1,17 @@
...
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { composeWithDevTools } from 'remote-redux-devtools';
 import { Provider } from 'react-redux';
 import thunk from 'redux-thunk';
-import debounce from 'lodash/debounce';
+import throttle from 'lodash/throttle';
 
 import Map from './glmap/Map.container';
-import { initModule } from './module/module.actions';
+import { initModule, setTemporalExtend } from './module/module.actions';
```
[the remark this answers is not in the document]
Comment (nerik) [reply app/src/_map/index.jsx]: Re

**Pull request merged**